src/nrniv/nrncore_write/callbacks/nrncore_callbacks.cpp

```diff
@@ -523,6 +550,37 @@ int core2nrn_corepointer_mech(int tid, int type, int icnt, int dcnt, int* iArray
     return 1;
 }
 
+// NMODL RANDOM seq34 data return from coreneuron
+int core2nrn_nmodlrandom(int tid,
+                         int type,
+                         std::vector<int>& indices,
+                         std::vector<double>& nmodlrandom) {
+    if (tid >= nrn_nthread) {
+        return 0;
+    }
+    NrnThread& nt = nrn_threads[tid];
+    Memb_list* ml = nt._ml_list[type];
+    // ARTIFICIAL_CELL are not in nt.
+    if (!ml) {
+        ml = CellGroup::deferred_type2artml_[tid][type];
+        assert(ml);
+    }
+
+    auto& nrnindices = nrn_mech_random_indices(type);  // for sanity checking
+    assert(nrnindices == indices);
+    assert(nmodlrandom.size() == indices.size() * ml->nodecount);
+
+    int ir = 0;  // into nmodlrandom
+    for (auto ix: nrnindices) {
```

Comment:
I'm not too picky about that, but I guess one could declare ix as `const auto` for such a foreach loop.

Reply:
will be ```const auto ix``` in next commit

Reply:
Done